tools/ratbagctl.body.py.in

```diff
@@ -0,0 +1,1889 @@
+    profile, device = find_profile(ratbagd, args)
+    profile.disabled = False
+    commit(device, args)
+
+
+def func_profile_disable(ratbagd: Ratbagd, args: argparse.Namespace) -> None:
+    profile, device = find_profile(ratbagd, args)
+    profile.disabled = True
+    commit(device, args)
+
+
+def func_device_name_get(ratbagd: Ratbagd, args: argparse.Namespace) -> None:
+    device = find_device(ratbagd, args)
+    print(device.name)
+
+
+################################################################################
+# these are definitions to be reused in the dict that defines our language
+
+# key elements
+"""the type of the element (see 'types' below)"""
+of_type = "type"
+"""the name of the element, it'll be the one matching the args on the CLI"""
+name = "name"
+"""the group to logically associate commands while printing the help"""
+group = "group"
+"""list of positional arguments for the given command"""
+pos_args = "pos_args"
+"""a tag that we can refer latrer in an element of type 'link'"""
+tag = "tag"
+"""the element pointed to in an element of type 'link'"""
+dest = "dest"
+"""the function to associate to the switch or command"""
+func = "func"
+"""this is a particular command that is an integer, but not an terminating argument.
+example:
+ profile active get
+ profile **2** button 3 get
+ - "profile" needs to be a switch
+ - "2" needs to be translated as a N_access, given it is a requirement to be able to call 'button'
+ """
+N_access = "N_access"
+
+# argparse.add_argument parameters (forwarded as such)
+"""'type' of the argument"""
+arg_type = "arg_type"
+"""'metavar' of the argument"""
+metavar = "metavar"
+"""'help' of the argument"""
+help_str = "help"
+"""'nargs' of the argument"""
+nargs = "nargs"
+"""'choices' of the argument"""
+choices = "choices"
+
+# types
+"""an option to interpret as a command (example 'list', 'info')"""
+command = "command"
+"""an argument that is required for the given command arguments are leaf nodes
+and can not have children
+"""
+argument = "argument"
+"""provides a list of choice of commands for instance, a switch of [A, B] means
+we can have A or B only when parsing the command line
+"""
+switch = "switch"
+"""same as list, except we can loop inside the list for instance, a set of
+[A, B] means we can have A and B (and A, ...) one after the other, no matter
+the order
+"""
+set = "set"
+"""a reference to any other element in the tree marked with a tag"""
+link = "link"
+
+################################################################################
+
+
+def commit(device: RatbagdDevice, args: argparse.Namespace) -> None:
+    if args.nocommit:
+        return
+    device.commit()
+
+
+def color(string: str) -> Tuple[int, int, int]:
+    try:
+        int_value = int(string, 16)
+    except ValueError as e:
+        msg = "%r is not a color in hex format" % string
+        raise argparse.ArgumentTypeError(msg) from e
```

Comment:
For this kind of errors, `ValueError` would be better, no?

Reply:
:+1: 
I thought it needed to be so because these function are used by argparse (I think) and it handled these somehow (it didn't).

Reply:
I checked source code of argparse, it actually does handle ArgumentTypeError specially, but so does it handle ValueError. Anyway it's better to use something more generic.